### PR TITLE
[WIP] getting the video links of explore page

### PR DESCRIPTION
**Pull request by akki111singh**
only taking some time.

### CODE REVIEW

**Comment by aashutoshrathi** [comment]
Hi @akki111singh 
It works but doesn't download content, instead opens in new tab.
Also, same thing can be fixed for profile section

**Comment by akki111singh** [comment]
@aashutoshrathi we can download the content in the same way as image,by right clicking and then save as

**Comment by aashutoshrathi** [comment]
Earlier there was download prompt, now that is not working

**Review by aashutoshrathi — COMMENTED** [review]
Also, this is some great work.
Refactor the code a bit and I think its good to merge

**Comment by akki111singh** [comment]
Added the video downloading feature in explore. other profile & self profile also faster than before by making async requests

**Review by aashutoshrathi — APPROVED** [review]
@akki111singh That looks great, is main page breaking on your PC too?

File: download.js [hunk 1]
```diff
@@ -4,150 +4,228 @@ link.type = "text/css";
...
+                y = images[i].parentElement.parentElement.parentElement;
+                getHTML(y, function(response) {
+                    var video = response.querySelectorAll("meta[property='og:video:secure_url']")
+                    let db = document.createElement("a");
 
-  // Grab images appearing on the page
-  videos = document.querySelectorAll("video");
-  let j = 0;
+                    db.innerHTML = `<a download href=${video[0].content}>
+                    <button class="instanshu-unite instanshu-sm instanshu-success" style="left: 100px; top: 10px;"> video Download</button>`;
```
Comment (aashutoshrathi) [inline download.js:89]: Make it `Download Video`

File: download.js [hunk 2]
```diff
@@ -4,150 +4,228 @@ link.type = "text/css";
...
+            let x = images[i].parentElement.parentElement.parentElement.parentElement;
+
+            if (x.getElementsByClassName("Byj2F").length === 1) {
+                y = images[i].parentElement.parentElement.parentElement;
+                getHTML(y, function(response) {
+                    var video = response.querySelectorAll("meta[property='og:video:secure_url']")
+                    let db = document.createElement("a");
+
+                    db.innerHTML = `<a download href=${video[0].content}>
+                    <button class="instanshu-unite instanshu-sm instanshu-success" style="left: 100px; top: 10px;"> video Download</button>`;
```
Comment (aashutoshrathi) [inline download.js:130]: Here too.

File: download.js [hunk 3]
```diff
@@ -4,150 +4,228 @@ link.type = "text/css";
...
+        for (let i = 1; i < images.length; i++) {
+            let x = images[i].parentElement.parentElement.parentElement.parentElement;
+            if (x.getElementsByClassName("Byj2F").length === 1) {
+                y = images[i].parentElement.parentElement.parentElement;
+                getHTML(y, function(response) {
+                    var video = response.querySelectorAll("meta[property='og:video:secure_url']")
+                    let db = document.createElement("a");
+
+                    db.innerHTML = `<a download href=${video[0].content}>
+                    <button class="instanshu-unite instanshu-sm instanshu-success" style="left: 100px; top: 10px;"> video Download</button>`;
```
Comment (aashutoshrathi) [inline download.js:167]: Same here

**Review by aashutoshrathi — COMMENTED** [review]
Check comments

**Comment by akki111singh** [comment]
@aashutoshrathi  Added downloading feature instead of opening for both the images and video

**Comment by aashutoshrathi** [comment]
Can you resolve conflicts and run eslint to lint the code.

**Comment by akki111singh** [comment]
@ how to run eslint it is throwing some error.

**Comment by akki111singh** [comment]
@aashutoshrathi  done eslint on download.js

**Review by aashutoshrathi — APPROVED** [review]
Looks good.
Should I merge it then?
Have you checked it locally?

File: download.js (hunk 1 above)
Comment (aashutoshrathi) [reply download.js:89]: This is done?

**Comment by akki111singh** [comment]
@aashutoshrathi Yes acc. to me the explore page,other profile & self profile is working nicely.You should check it once.
Now the main page is broken also in case of multiple photos only one can download the first image.

**Comment by akki111singh** [comment]
@aashutoshrathi main page must be improved as it doesn't look nice with buttons here and there also the video part in main page is not working only thumbnail is downloading.

**Comment by aashutoshrathi** [comment]
@akki111singh I'm merging this, let's take care of main page in another PR.

**Comment by akki111singh** [comment]
@aashutoshrathi i will work on main page + what kind of enhancements are required in popup.html

**Comment by aashutoshrathi** [comment]
Someone is working on PR.

**Comment by akki111singh** [comment]
@aashutoshrathi any other issue i can work on?

**Comment by aashutoshrathi** [comment]
You can take main page PR, if you want.
Just align fix the video download on main page and multiple images, I'm planning to change button style too.

**Pull request merged**